Assets/Scripts/ArenasParametersSideChannel.cs

```
@@ -17,11 +20,54 @@ public ArenasParametersSideChannel()
 
     protected override void OnMessageReceived(IncomingMessage msg)
     {
-        byte[] yamlData = msg.GetRawBytes();
+        byte[] rawData = msg.GetRawBytes();
+        string potentialFilePath = Encoding.UTF8.GetString(rawData);
 
-        /* Create the event args and the YAML data */
-        ArenasParametersEventArgs args = new ArenasParametersEventArgs { arenas_yaml = yamlData, };
-        OnArenasParametersReceived(args);
+        // Check if the received string is a valid file path
+        if (!string.IsNullOrEmpty(potentialFilePath) && File.Exists(potentialFilePath))
+        {
+            // New behavior: read arena config from file
+            byte[] yamlData = ReadArenaConfigFile(potentialFilePath);
+            if (yamlData != null)
```

When might yamlData be null, and what would happen in that case?

---

The yaml data could be null if the file path given is incorrect or the file cannot be read for some reason or the file is empty. If it is null then the arena isn't loaded and and error is logged. Would you prefer that if the yamml data is null it throws the error fully and crashes?

---

Which error is logged here? I'm it might be worth having an explicit else branch on this with a `Debug.LogError` to make the failure really traceable (regrettably following the codebase silent failure standard), but maybe the current error is fine